ambuda/models/auth.py

```diff
@@ -0,0 +1,90 @@
+from flask_login import UserMixin
+from enum import Enum
+from datetime import datetime
+from sqlalchemy import (
+    Column,
+    DateTime,
+    Integer,
+    String,
+)
+from sqlalchemy.orm import relationship
+from werkzeug.security import generate_password_hash, check_password_hash
+
+from ambuda.models.base import Base, foreign_key, pk
+
+
+class SiteRole(Enum):
+    PROOFREADER = "Allow proofreading of texts prior."
+    SITE_ADMIN = "Allow management of site data."
+
+
+class UserRoles(Base):
+
+    """Roles for each user"""
+
+    __tablename__ = "user_roles"
+
+    #: Primary key.
+    id = pk()  # TODO: make pk both columns
+    #: The user that has this permission
+    user_id = foreign_key("users.id")
+    #: The user that has this permission
+    role_id = foreign_key("roles.id")
+
+    #: The user that this role belongs to.
+    user = relationship("User")
+    #: The role that this user has.
+    role = relationship("Role")
+
+
+class Role(Base):
+
+    """A role"""
+
+    __tablename__ = "roles"
+
+    #: Primary key.
+    id = pk()
+    #: Name of the role name
+    name = Column(String, unique=True, nullable=False)
+    #: Description of the role
+    description = Column(String, nullable=False)
+    #: When this permission was added
+    creation_time = Column(DateTime, nullable=False, default=datetime.utcnow)
```

Comment:
I've been using `created_at` since it's a little shorter. Let's standardize on that.